### PR TITLE
Update spark_config to handle k8s container required volumes for paasta spark-run

### PR DESCRIPTION
**Pull request by gcollura**
Change related to https://github.com/Yelp/paasta/pull/3174

## What changed
- [x] driver containers were missing essentials volumes when k8s was used as cluster manager, this review matches the behavior for the mesos CM.

### CODE REVIEW

File: service_configuration_lib/spark_config.py
Comment (gcollura) [inline service_configuration_lib/spark_config.py:186]: This is not great, hence the WIP

**Comment by 88manpreet** [comment]
Please also add @vkhromov as primary reviewer.

File: service_configuration_lib/spark_config.py
Comment (88manpreet) [reply service_configuration_lib/spark_config.py:186]: +1 for just these 2 variables, I would just add it manually. Looks less complicated.
What was the error you were getting if this is not mounted?

Comment (88manpreet) [reply service_configuration_lib/spark_config.py:186]: Or make it similar to what is being done for DEFAULT_SPARK_K8S_SECRET_VOLUME

**Comment by 88manpreet** [comment]
Can you point to the similar code for mesos CM? 
Ref.:  driver containers were missing essentials volumes when k8s was used as cluster manager, this review matches the behavior for the mesos CM.
How did it work on k8s job deployed on tron?

**Comment by 88manpreet** [comment]
lgtm
Looks like the final testing would be done after changes are merged.
The changes look safe to me for testing purposes.

**Review by 88manpreet — APPROVED** [review]

**Review by chelspea — COMMENTED** [review]
LGTM

File: service_configuration_lib/spark_config.py
Comment (88manpreet) [inline service_configuration_lib/spark_config.py:435]: I also created the PR for [this](https://github.com/Yelp/service_configuration_lib/pull/65). 
It's good it's been done as part of single PR, I will discard mine.

File: service_configuration_lib/spark_config.py
Comment (88manpreet) [inline service_configuration_lib/spark_config.py:68]: Is it possible to keep K8S_AUTH_FOLDER same as hostPath everywhere?
I personally prefer keeping things consistent (here hostPath == containerPath) as much as possible for simplicity (debugging, memorization, ease manual testing).

**Review by 88manpreet — APPROVED** [review]
Nit comments. Otherwise, looks good to me. 
I believe @vkhromov has been added in the review primarily for visibility or any review-comments related how spark-infra interacts with k8s?

@vkhromov kindly let us know if we need to get blocked on your review-comments.

**Review by 88manpreet — APPROVED** [review]

File: service_configuration_lib/spark_config.py
Comment (88manpreet) [inline service_configuration_lib/spark_config.py:177]: I just realized this is confusing. 
For consistency, let's either remove DEFAULT_SPARK_K8S_SECRET_VOLUME to change to format k8s_volumes.append({'containerPath': '/etc/group', 'hostPath': '/etc/group', 'mode': 'RO'})
or the other way around?

Or is there a reason for two different patterns for appending?

**Pull request merged**
